alicloud/data_source_alicloud_disks.go

```diff
@@ -0,0 +1,273 @@
+			},
+			"category": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDiskCategory,
+			},
+			"encrypted": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"tags": tagsSchema(),
+			"output_file": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			// Computed values
+			"disks": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"region_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"availability_zone": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"category": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"encrypted": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"image_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"instance_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"creation_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"attached_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"detached_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"expiration_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tags": tagsSchema(),
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAlicloudDisksRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+
+	args := ecs.CreateDescribeDisksRequest()
+
+	if v, ok := d.GetOk("ids"); ok && len(v.([]interface{})) > 0 {
+		args.DiskIds = convertListToJsonString(v.([]interface{}))
+	}
+	if v, ok := d.GetOk("type"); ok && v.(string) != "" {
+		args.DiskType = v.(string)
+	}
+	if v, ok := d.GetOk("category"); ok && v.(string) != "" {
+		args.Category = v.(string)
+	}
+	if v, ok := d.GetOk("encrypted"); ok && v.(string) != "" {
+		if v == "on" {
+			args.Encrypted = requests.NewBoolean(true)
+		} else if v == "off" {
+			args.Encrypted = requests.NewBoolean(false)
+		}
```

Comment:
I think use a const: https://github.com/alibaba/terraform-provider/blob/master/alicloud/extension_slb.go#L17 instead 'on' or 'off' will be better.